test/unit/Factory.t.sol

```diff
@@ -0,0 +1,349 @@
+    /// VERSION() function but contain a fallback()
+    /// function and AccountFailedToFetchVersion error
+    /// would *NOT* be triggered.
+    ///
+    /// Given this, it is up to the factory owner to take
+    /// extra care when creating the implementation to be used
+    function testWhenAccountCannotFetchVersion() public {
+        MockAccount2 mockAccount = new MockAccount2();
+
+        factory = new Factory({
+            _owner: address(this),
+            _marginAsset: SUSD,
+            _addressResolver: ADDRESS_RESOLVER,
+            _settings: address(settings),
+            _ops: payable(GELATO_OPS),
+            _implementation: address(mockAccount)
+        });
+
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                IFactory.AccountFailedToFetchVersion.selector,
+                ""
+            )
+        );
+        factory.newAccount();
+    }
+
+    function testNewAccountEvent() public {
+        vm.expectEmit(true, false, false, false);
+        emit NewAccount(address(this), address(0), bytes32(0));
+        factory.newAccount();
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                           ACCOUNT OWNERSHIP
+    //////////////////////////////////////////////////////////////*/
+
+    function testAccountCanTransferAccountOwnership() public {
+        address payable accountAddress = factory.newAccount();
+        Account(accountAddress).transferOwnership({
+            _newOwner: address(0xCAFEBAE)
+        });
+        assertEq(factory.ownerToAccount(address(this)), address(0));
+        assertEq(factory.ownerToAccount(address(0xCAFEBAE)), accountAddress);
+        assertEq(Account(accountAddress).owner(), address(0xCAFEBAE));
+    }
+
+    function testAccountCannotTransferOwnershipToAnotherAccountOwningAddress()
+        public
+    {
+        address payable accountAddress1 = factory.newAccount();
+        vm.prank(KWENTA_TREASURY);
+        address payable accountAddress2 = factory.newAccount();
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                IFactory.OnlyOneAccountPerAddress.selector,
+                accountAddress2
+            )
+        );
+        Account(accountAddress1).transferOwnership({
+            _newOwner: KWENTA_TREASURY
+        });
+    }
+
+    function testCannotUpdateAccountThatDoesNotExist() public {
+        vm.expectRevert(
+            abi.encodeWithSelector(IFactory.AccountDoesNotExist.selector)
+        );
+        factory.updateAccountOwner({
+            _oldOwner: address(0xCAFEBAE),
+            _newOwner: address(0xBEEF)
+        });
+    }
+
+    function testCannotDirectlyUpdateAccount() public {
+        factory.newAccount();
+        vm.expectRevert(
+            abi.encodeWithSelector(IFactory.CallerMustBeAccount.selector)
+        );
+        factory.updateAccountOwner({
+            _oldOwner: address(this),
+            _newOwner: address(0xBEEF)
+        });
+    }
```

Comment:
Another good test is `testAccountOwnerCannotTransferAnotherAccount`

Reply:
Will add

Reply:
🏁🏁🏁